test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java

```diff
@@ -195,7 +198,21 @@ private ExtensionState doJavaStart(ExtensionContext context, Class<? extends Qua
 
             // If gradle project running directly with IDE
             if (System.getProperty(BootstrapConstants.SERIALIZED_APP_MODEL) == null) {
-                BuildToolHelper.enableGradleAppModelForTest(projectRoot);
+                QuarkusModel model = BuildToolHelper.enableGradleAppModelForTest(projectRoot);
+                final Set<File> classDirectories = model.getWorkspace().getMainModule().getSourceSet().getSourceDirectories();
+                for (File classes : classDirectories) {
+                    if (!rootBuilder.contains(classes.toPath())) {
+                        rootBuilder.add(classes.toPath());
+                    }
+                }
+            } else if (System.getProperty(BootstrapConstants.OUTPUT_SOURCES_DIR) != null) {
+                final String[] sourceDirectories = System.getProperty(BootstrapConstants.OUTPUT_SOURCES_DIR).split(",");
```

Comment:
If a project has only one output test dir, wouldn't this dir be added twice? I.e. through this property and the default test dir that would have been added w/o this code?

Reply:
Yes, that's ok.

However, could you remind me why we did `BuildToolHelper.enableGradleAppModelForTest(projectRoot)`? Is this to support IDEs?

Reply:
`BuildToolHelper.enableGradleAppModelForTest(projectRoot)` is already checking whether it's a gradle project, btw.

Reply:
Yes exactly if you come from the IDE the model has not been serialized yet. This is why there is those two similar blocks. 
I added the check in the `if` because the `BuildToolHelper.enableGradleAppModelForTest(projectRoot)` returns `null` if it's not a Gradle project. You are right, it's better to check if the returned model is not null rather than checking if this a Gradle project (as the method recursively look for a build file). I will update the code